tasties_app/views.py

```diff
@@ -19,12 +19,39 @@ def base(request):
 
 def recipes(request):
     recipes_list = Recipe.objects.all()
+    categories_list = Category.objects.all()
+    selected_category = request.GET.get("category")
     recipes_with_ratings = {}
+
+    if selected_category != 'remove_filter' and selected_category:
+        for recipe in recipes_list:
+            categories_in_recipe = recipe.categories.all()
+            for category in categories_in_recipe:
+                if category.category_name in selected_category:
+                    recipes_with_ratings[recipe] = Rating.objects.filter(
+                        recipe_id=recipe
+                    ).aggregate(Avg("rating"))["rating__avg"]
+        recipes_with_ratings = OrderedDict(
+            sorted(recipes_with_ratings.items(), key=lambda x: x[1], reverse=True)
+        )
+        context = {
+            "recipes_with_ratings": recipes_with_ratings,
+            "categories_list": categories_list,
+        }
+        return render(request, "tasties_app/recipes.html", context)
```

Comment:
You have a duplicate code here, I think you can make the code more readable if you remove lines 37-41 (they are the same as lines 50-54) and put lines 43-49 under `else` block.